bin/src/hickory-dns.rs

```diff
@@ -925,3 +929,68 @@ fn build_udp_socket(ip: IpAddr, port: u16) -> Result<UdpSocket, Error> {
 
     UdpSocket::from_std(sock.into())
 }
+
+/// Drop privileges on Unix systems if running as root. Errors that prevent dropping privileges will
+/// halt the server.  This must be called after binding to low numbered sockets is complete.
```

Comment:
To clean up the function a bit, I did change it to return a Result with the error condition, but the call to the function propagates the error up and still triggers an exit.

This is what we do with other configuration errors, and I think that is the right thing to do, particularly as it relates to global settings like port bindings and the system privileges to run the daemon as.   If we can't configure the daemon as the configuration file dictates, that, to me is an error the user should have to correct proactively rather than us try to proceed with a half-configured service.

Reply:
Oh yeah, it makes sense to me that we want to exit the process, I just think it's cleaner/less surprising to do it by yielding an error... as you did.